File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.py
def solution(n):
    answer = ''
    nums = ["4", "1", "2"]
    
    while n >= 1:
        c = n % 3
        answer = nums[c] + answer
        n -= c if c != 0 else 3
        n //= 3
        
    
    return answer

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
class Solution {
    public String solution(int n) {
        StringBuilder sb = new StringBuilder();
        String[] nums = new String[]{"4", "1", "2"};
        
        while (n > 0) {
            int c = n % 3;
            sb.append(nums[c]);
            n -= c != 0 ? c : 3;
            n /= 3;
        }
        
        return sb.reverse().toString();
    }
}